dist/linux/travis-ci.sh

```diff
@@ -1,12 +1,65 @@
-IFS='+' read -ra REFS <<< "$BLD"
+#! /bin/sh
+# This script is executed in the travis-ci environment.
 
-DBLD=${REFS[1]}
-PKG_DTAG=${REFS[0]}
-PKG_SHORTCOMMIT="$(echo $TRAVIS_COMMIT | cut -c1-10)"
+thismatrix=regular
+
+images=("ghdl/ghdl-tools:ubuntu-mcode"    \
+       "ghdl/ghdl-tools:ubuntu1404-llvm" \
+	   "ghdl/ghdl-tools:ubuntu1204-llvm" \
+       "ghdl/ghdl-tools:fedora-llvm-mcode")
+
+#--- image + identifier + compiler
+	   
+regular=("0+ubuntu+mcode" \
+         "3+fedora+llvm")
+
+nightly=("1+ubuntu1404+llvm-3.5" \
+         "2+ubuntu1204+llvm-3.8" \
+		 "3+fedora+mcode")
+
+#release=()
+
+#---
+
+task() {
+  printf "\033[33m[$1| BUILD] $2 \033[0m\n"
+
+  IFS='+' read -ra REFS <<< "$2"
+  DBLD=${REFS[2]}
+  wrkd="../wrk-$m"
+  cp -r ./ "$wrkd" && cd "$wrkd"
+  ./dist/linux/docker-buildtest.sh -i "${images[${REFS[0]}]}" -t "$1" -b "$DBLD" -f "ghdl-$PKG_VER-$DBLD-$PKG_TAG-${REFS[1]}-$PKG_SHORTCOMMIT.tgz"
+  cd "$cloned"
+  cp $(ls "$wrkd"/ghdl-*.tgz) ./
+  cp $(ls "$wrkd"/log.log) "./log_$m.log"
+}
+
+#---
+
+PKG_SHORTCOMMIT="$(printf $TRAVIS_COMMIT | cut -c1-10)"
 PKG_VER=`grep Ghdl_Ver src/version.in | sed -e 's/.*"\(.*\)";/\1/'`
 PKG_TAG="$TRAVIS_TAG"
-
-if [ -z "$BUILDTHIS" ]; then BUILDTHIS=$(echo "$TRAVIS_TAG"); fi
 if [ -z "$TRAVIS_TAG" ]; then PKG_TAG=`date -u +%Y%m%d`; fi
 
-export PKG_FILE="ghdl-$PKG_VER-$DBLD-$PKG_TAG-$PKG_DTAG-$PKG_SHORTCOMMIT.tgz"
\ No newline at end of file
+cloned=$(pwd)
+
+printf "\033[33m[TRAVIS] Running matrix $thismatrix \033[0m\n"
+eval blds=( '"${'${thismatrix}'[@]}"' )
+for m in $(eval echo "{0..$((${#blds[@]}-1))}"); do
```

Comment:
This minimum example fails both in MinGW and in travis-ci:
```
blds=("0+ubuntu+mcode" "3+fedora+llvm")
for m in {0..$((${#blds[@]}-1))}; do echo "$m" "${blds[$m]}"; done
```
That's because braces are evaluated before the content between them. The eval avoids that:
```
for m in $(eval echo "{0..$((${#blds[@]}-1))}"); do echo "$m" "${blds[$m]}"; done
```
And according to http://unix.stackexchange.com/questions/55392/in-bash-is-it-possible-to-use-an-integer-variable-in-the-loop-control-of-a-for an alternative is to use a C-style for loop. But I don't know if that will affect portability.